catalyst-generator/src/main/java/com/intellecteu/catalyst/generator/ProjectGenerator.java

```diff
@@ -260,7 +257,10 @@ protected File generateProjectStructure(ProjectRequest request,
       writeGradleWrapper(dir, Version.safeParse(request.getBootVersion()));
     } else {
       String pom = new String(doGenerateMavenPom(model));
-      writeText(new File(dir, "pom.xml"), pom);
+      StringBuilder pomBuilder = new StringBuilder(pom);
+      int idx = pomBuilder.indexOf("{customPlugins}");
+      pomBuilder.replace(idx, idx+15, resolveCustomPlugins(request, 3));
```

Comment:
Why do you use string replace for this case? You can add a model variable (and a placeholder in pom), and process it using normal doGenerateMavenPom(model) method used (3 lines above)